folium/features.py

```diff
@@ -352,6 +410,13 @@ def style_data(self):
 
         for feature in self.data['features']:
             feature.setdefault('properties', {}).setdefault('style', {}).update(self.style_function(feature))  # noqa
+
+            try:
+                for feature in self.data["features"]:
+                    feature["properties"]["_popupContent"] = self.popup_function(feature)  # noqa
+            except TypeError:
+                pass
+
```

Comment:
Since the checks on the function are outside the render operation, I was being a bit slack and letting the function fail when it wasn't a function. I'll move checks inside the loop.